.github/workflows/manual_firmware_build.yml

```diff
@@ -0,0 +1,75 @@
+name: Manual Firmware build (produce downloadable *_with_bl.hex ready to upload to FC)
+
+on:
+  workflow_dispatch:
+    inputs:
+      boards:
+        description: 'Boards to build (see `Tools/scripts/board_list.py` for a full list)'
+        required: true
+        default: 'MatekF405-Wing,MatekF405-CAN,MatekF765-Wing,MambaF405v2'
+      projects:
+        description: 'Projects to build (e.g. `arducopter,arduplane,rover,antennatracker,ardusub,AP_Periph` )'
+        required: true
+        default: 'arduplane'
+
+concurrency:
+  group: ci-${{github.workflow}}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: 'macos-latest'
```

Comment:
why macos ? that is the worst GH machine to do that

Reply:
I guess this is because all other existing workflows are named `test_*.yml`  and only for cygwin & macos was named `build_*.yml` (i.e. no`build_linux.yml` or so), so seemed as quick starting point.  https://github.com/ArduPilot/ardupilot/tree/master/.github/workflows

not sure if some file rename or extra developer docs would help?